auth-layer-proxy/README.md

```diff
@@ -0,0 +1,184 @@
+
+```mermaid
+---
+title: Overall Architecture
+---
+flowchart LR
+    A[Graph Cli] --->|1. Create / Deploy Subgraph| B(Auth-Layer)
+    B ---> | 4. Relay Request on Admin Port| F(TheGraph Service)
+    B --> | 3. 401 Unauthorized | A
+    F ---> | 5. response| B --> | 6. response| A
+    B <--> | 2. Check Token| db[(Database)]
+    
+    linkStyle 2 stroke:#ff0000,stroke-width:2px;
+    linkStyle 3 stroke:#00ff00,stroke-width:2px;
+    linkStyle 4 stroke:#00ff00,stroke-width:2px;
+
+```
+More information on the **Authorization Layer** can be found [here](https://github.com/hashgraph/hedera-the-graph/blob/main/docs/design/auth-layer.md)
+
+## Overview
+
+This is an implementation of EnvoyProxy filters for authentication and authorization. It is a custom config with http filters using Lua scripts for performing the following actions: 
+
+1. JSON Validation
+2. Token Extraction
+4. Payload Params Extraction
+5. Token Validation using JWT and Instrospect endpoint
+6. Proxy Routing Configuration (using EnvoyProxy itself)
+
+it includes a Dockerfile for building the image and a docker-compose file for running the container.
+
+## Pre-requisites
+
+### OAUTH 2.0 Token Server
+This auth-token validation proxy layer relies on an OAuth 2.0 token server for token issuace and validation. The token server should be able to issue and validate the token using the `client_id` and `client_secret` provided in the `.env` file.
+
+So make sure to have a token server running that is previously configured with a Client ID and Client Secret, and the `/token` and `/token/introspection` endpoints are accessible.
+
+### Token structure
+
+Make sure that the access token has the following claims:
+  
+  ```json
+  {
+    "exp": 1711427468,
+    "iat": 1711391468,
+    "jti": "2fab170f-beb1-4821-acb4-ac19a71c9abe",
+    "iss": "http://host.docker.internal:8080/realms/HederaTheGraph",
+    "aud": "account",
+    "sub": "f60ffb03-d17f-4aa2-a34a-2c4891059c3c",
+    "typ": "Bearer",
+    "azp": "htg-auth-layer",
+    "session_state": "79fbb78a-3279-463e-8ee0-6ab37e06bcc2",
+    "acr": "1",
+    "allowed-origins": [
+        "/*"
+    ],
+    "realm_access": {
+        "roles": [
+            "default-roles-hederathegraph",
+            "subgraph_remove",
+            "subgraph_create",
+            "subgraph_deploy",
+            "offline_access",
+            "uma_authorization",
+            "subgraph_resume",
+            "subgraph_pause"
+        ]
+    },
+    "scope": "profile subgraph_access email",
+    "subgraph_access": "<CSV of subgraph names>",
+    "email_verified": true,
+    "active": true,
+    "email": "user1@gmail.com",
+    "client_id": "htg-auth-layer"
+  }
+  ```
+
+For instructions on how to set-up the Auth Provider using KeyCloak, refer to the `Auth-Layer-Server` [README](link)
```

Comment:
Update link

Reply:
It was a placeholder for when the auth-layer-server was already merged, now that is merged I have updated with the approapiate URL. thanks for noticing!!

✅ Done